File: PageObjectJava/src/main/java/Support/Support.java
package Support;

import Base.TestConfiguration;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.openqa.selenium.*;
import org.openqa.selenium.interactions.Actions;
import ru.yandex.qatools.ashot.AShot;
import ru.yandex.qatools.ashot.Screenshot;
import ru.yandex.qatools.ashot.shooting.ShootingStrategies;

import java.util.*;

public class Support {

    public static final Logger LOGGER = LogManager.getLogger("selenium");
    private static final List<Screenshot> SAVED_SCREENSHOTS = new ArrayList<>();

    public static void javaScriptClick(WebElement element){
        getJavaScriptExecutor().executeScript("arguments[0].click();", element);
    }

    public static String extractTextWithJavaScript(WebElement element){
        return getJavaScriptExecutor()
                .executeScript("return arguments[0].textContent;", element)
                .toString();
    }

    public static JavascriptExecutor getJavaScriptExecutor(){
        return (JavascriptExecutor) TestConfiguration.getDriver();
    }

    public static void takeScreenshot(){
        takeScreenshot(null);
    }

    public static void takeScreenshot(WebElement webItem){
        AShot shotInstance = new AShot().shootingStrategy(ShootingStrategies.viewportPasting(1000));

        Screenshot shot = Objects.isNull(webItem) ? shotInstance.takeScreenshot(TestConfiguration.getDriver()) :
                shotInstance.takeScreenshot(TestConfiguration.getDriver(), webItem);
        SAVED_SCREENSHOTS.add(shot);
    }

    public static void HoverOverWebElementsThenClickOver(WebElement finalWebItem, WebElement... orderedWebItems)
    {
        Actions act = new Actions(TestConfiguration.getDriver());

        for (WebElement itm : orderedWebItems)
            act = act.moveToElement(itm);

        act.click(finalWebItem).build().perform();
    }

    public static void hoverOverWebElement(WebElement element){
        scrollToWebElement(element);
        Actions act = new Actions(TestConfiguration.getDriver());
        act.moveToElement(element).build().perform();
    }

    public static void scrollToWebElement(WebElement element)
    {
        JavascriptExecutor executor = getJavaScriptExecutor();
        executor.executeScript("arguments[0].scrollIntoView();", element); //Necessary for Firefox
    }

    public static List<Screenshot> getStoredScreenshots(){
        return SAVED_SCREENSHOTS;
    }

    public static void clearScreenshotsList(){
        SAVED_SCREENSHOTS.clear();
    }

}

File: PageObjectDotNet/SeleniumFramework/Base/Support.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Interactions;
using System;
using System.IO;
using System.Reflection;

namespace SeleniumFramework.Base
{
    public static class Support
    {
        public static readonly log4net.ILog Logger = log4net.LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);

        /// <summary>
        /// This method takes a screenshot, and returns the fullpath of the file saved
        /// </summary>
        /// <param name="_driver"></param>
        /// <param name="SectionName"></param>
        /// <returns>The taken screenshot path</returns>
        public static string TakeScreenShoot(this IWebDriver _driver, string SectionName, string sreenShotPath)
        {
            var baseScreenShotLocation = sreenShotPath;
            if (!Directory.Exists(baseScreenShotLocation))
                Directory.CreateDirectory(baseScreenShotLocation);

            var SSName = Path.Combine(baseScreenShotLocation, SectionName + "_" + DateTime.Now.ToFileTimeUtc() + ".png");
            Screenshot ss = ((ITakesScreenshot)_driver).GetScreenshot();
            ss.SaveAsFile(SSName, ScreenshotImageFormat.Png);
            return SSName;
        }

        /// <summary>
        /// This method returns a <see cref="IJavaScriptExecutor"/> to support Javascript code execution
        /// </summary>
        /// <param name="driver"></param>
        /// <returns></returns>
        public static IJavaScriptExecutor Scripts(this IWebDriver _driver)
        {
            var js = _driver as IJavaScriptExecutor;
            return js;
        }

        /// <summary>
        /// This method highlights an element on the Site using a given locator
        /// </summary>
        /// <param name="driver"></param>
        /// <param name="locator"></param>
        public static void HighlightElement(this IWebDriver driver, By locator)
        {
            HighlightElement(driver, driver.FindElement(locator));
        }


        /// <summary>
        /// This method highlights an element on the Site using the given webelement
        /// </summary>
        /// <param name="driver"></param>
        /// <param name="element"></param>
        public static void HighlightElement(this IWebDriver driver, IWebElement element)
        {
            var js = Scripts(driver);

            string highlightJavascript = @"arguments[0].style.cssText = ""border-width: 3px; border-style: solid; border-color: red""; ";
            js.ExecuteScript(highlightJavascript, element);
            highlightJavascript = @"arguments[0].style.cssText = ""border-width: 0px"";";
            js.ExecuteScript(highlightJavascript, element);
        }

        public static void JavaScriptClick(this IWebDriver driver, IWebElement elementToClick)
        {
            var js = Scripts(driver);
            js.ExecuteScript(@"arguments[0].click();", elementToClick);
        }

        public static string GetTextUsingJavaScript(this IWebDriver driver, IWebElement elementToExtractText)
        {
            var js = Scripts(driver);
            var obj = js.ExecuteScript(@"return arguments[0].textContent;", elementToExtractText);
            return obj.ToString().Trim();
        }

        public static void HoverOverWebElement(this IWebDriver driver, IWebElement element) 
        {
            ScrollToWebElement(driver, element);
            var act = new Actions(driver);
            act.MoveToElement(element).Build().Perform();
        }

        public static void ScrollToWebElement(this IWebDriver driver, IWebElement element) 
        {
            var js = Scripts(driver);
            js.ExecuteScript("arguments[0].scrollIntoView();", element);
        }


    }
}
